src/prefect/cli/deployment.py

```diff
@@ -516,12 +517,29 @@ async def run(
         None,
         "--start-at",
     ),
+    watch: bool = typer.Option(
+        None,
+        "--watch",
+        help="Whether to poll the flow run until a terminal state is reached.",
+    ),
```

Comment:
I recommend defaulting to `False` since it seems to mean the same as `None` in this context.
```suggestion
    watch: bool = typer.Option(
        False,
        "--watch",
        help="Whether to poll the flow run until a terminal state is reached.",
    ),
```